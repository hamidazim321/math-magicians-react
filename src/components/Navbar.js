import { NavLink } from 'react-router-dom';
import '../styles/Navbar.css';

const Navbar = () => (
  <nav>
    <p className="brandName">Math Magicians</p>
    <ul>
      <li>
        <NavLink className="links" to="/"> Home </NavLink>
      </li>
      <li>
        <NavLink className="links" to="/calculator"> Calculator </NavLink>
      </li>
      <li>
        <NavLink className="links" to="/quotes"> Quotes </NavLink>
      </li>
    </ul>
  </nav>
);
export default Navbar;
