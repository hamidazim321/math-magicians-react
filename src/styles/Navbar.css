nav {
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 0 20px;
}

.brandName {
    font-size: 32px;
    font-weight: 600;
} 

ul {
    display: flex;
    gap: 10px;
    list-style-type: none;
}

li {
    border: solid 1px;
    color: black;
    border-top: none;
    border-bottom: none;
    padding: 0 5px;
}

.links {
    text-decoration: none;
    font-size: 24px;
    font-weight: 600;
}