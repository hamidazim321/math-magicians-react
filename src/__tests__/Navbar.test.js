import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom'; // Import MemoryRouter
import Navbar from '../components/Navbar';

test('Navbar component matches snapshot', () => {
  const { asFragment } = render(
    <MemoryRouter>
      <Navbar />
    </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});
