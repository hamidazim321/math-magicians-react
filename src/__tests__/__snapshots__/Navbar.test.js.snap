// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component matches snapshot 1`] = `
<DocumentFragment>
  <nav>
    <p
      class="brandName"
    >
      Math Magicians
    </p>
    <ul>
      <li>
        <a
          aria-current="page"
          class="links active"
          href="/"
        >
           Home 
        </a>
      </li>
      <li>
        <a
          class="links"
          href="/calculator"
        >
           Calculator 
        </a>
      </li>
      <li>
        <a
          class="links"
          href="/quotes"
        >
           Quotes 
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
